### PR TITLE
Update README for selection and filter

### PR DESCRIPTION
**Pull request by kjmahalingam**
There was limited description of the functionality of the primary selection strategy and the filter strategy in the config file. This PR adds that detail.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,8 +135,8 @@ The contents of the `implementationGuide` object's `primarySelectionStrategy` ob
 
 |Parameter |Type    |Description                                                                                                 |
 |:---------|:-------|:-----------------------------------------------------------------------------------------------------------|
-|`strategy`|`string`|The strategy to follow for primary selection (`"namespace"` or default `"entry"`).                          |
-|`primary` |`[]    `|An array of strings containing the namespaces to select as primary (only used for `"namespace"` `strategy`).|
+|`strategy`|`string`|The strategy to follow for primary selection (`"namespace"`, `"hybrid"`, or default `"entry"`).|
+|`primary` |`[]    `|An array of strings containing the namespaces to select as primary (only used for `"namespace"` and `"hybrid"` `strategy`).|
```
Comment (cmoesel) [inline README.md]: This isn't _just_ namespaces.  For `hybrid` strategy, it might also be entries.

File: README.md [hunk 2]
```diff
@@ -146,6 +146,22 @@ The contents of the `filterStrategy` object are as follows:
 |`strategy`|`string` |The strategy for specification filtering (`"namespace"`, `"element"`, or `"hybrid"`).|
 |`target`  |`[]`     |An array of strings containing the names for what to filter.                         |
 
+# Primary Selection Strategy and Filter Strategy
+
+The options for the configuration file's `implementationGuide.primarySelectionStrategy` are described below.
+
+* The `"entry"` `strategy` for primary selection sets every entry as primary in the IG.
+* The `"namespace"` `strategy` for primary selection sets every entry found in the namespaces in the `primary` array as primary in the IG.
+* The `"hybrid"` `strategy` for primary selection sets every entry listed in the `primary` array or found in the namespaces in the `primary` array as primary in the IG.
```
Comment (cmoesel) [inline README.md]: We may need/want an example of how an `entry` is specified (e.g., using dot notation?)

File: README.md [hunk 3]
```diff
@@ -146,6 +146,22 @@ The contents of the `filterStrategy` object are as follows:
...
+* If there is no `strategy` set in the `implementationGuide.primarySelectionStrategy`, the default operation is the `"entry"` `strategy`.
+
+The options for the configuration file's `implementationGuide.filterStrategy` are described below.
+
+* The `"element"` `strategy` for filtering will filter the specifications to only include the elements listed in the `target` array and their recursive dependencies.
+* The `"namespace"` `strategy` for filtering will filter the specifications to only include the elements included in the namespaces listed in the `target` array and their recursive dependencies.
+* The `"hybrid"` `strategy` for filtering will filter the specifications to only include the elements listed in the `target` array and included in the namespaces listed in the `target` array and their recursive dependencies.
+* If there is no `implementationGuide.filterStrategy` set, filtering will not occur.
```
Comment (cmoesel) [inline README.md]: Should you also briefly describe the `filter` flag?

**Comment by kjmahalingam** [comment]
Thanks for the input, @cmoesel . I've updated the README based on these suggestions.

File: README.md [hunk 4]
```diff
@@ -146,6 +146,27 @@ The contents of the `filterStrategy` object are as follows:
...
+
+The options for the configuration file's `implementationGuide.filterStrategy` are described below.
+
+* The `"element"` `strategy` for filtering will filter the specifications to only include the elements listed in the `target` array and their recursive dependencies.
+* The `"namespace"` `strategy` for filtering will filter the specifications to only include the elements included in the namespaces listed in the `target` array and their recursive dependencies.
+* The `"hybrid"` `strategy` for filtering will filter the specifications to only include the elements listed in the `target` array and included in the namespaces listed in the `target` array and their recursive dependencies.
+* If `filter` is `true`, then the filtering operation will occur. Otherwise, no filtering will occur.
+* If there is no `implementationGuide.filterStrategy` set, filtering will not occur.
+
+When specified a namespace or element in the `primary` or `target` array of either strategy, it is best
```
Comment (cmoesel) [inline README.md]: Should this be "When specifying"?  It reads kind of funny as-is.

Comment (kjmahalingam) [reply README.md]: Yup, that's a typo. Fixing now.

**Comment by markkramerus** [comment]
Thank you for addressing this, Dylan!

**Pull request merged**